src/tests/networking_tests.rs

```diff
@@ -692,30 +693,44 @@ pub mod net_tests {
         assert_eq!(cage.bind_syscall(serversockfd, &socket), 0);
         assert_eq!(cage.listen_syscall(serversockfd, 4), 0);
 
-        let inputs = interface::RustHashSet::<i32>::new();
-        let outputs = interface::RustHashSet::<i32>::new();
-        let excepts = interface::RustHashSet::<i32>::new();
+        // allocate spaces for fd_set bitmaps
+        let mut input_chunk: [u8; 128] = [0; 128];
+        let mut output_chunk: [u8; 128] = [0; 128];
+        let mut except_chunk: [u8; 128] = [0; 128];
 
-        inputs.insert(serversockfd);
-        inputs.insert(filefd);
-        outputs.insert(filefd);
+        let inputs: *mut u8 = input_chunk.as_mut_ptr();
+        let outputs: *mut u8 = output_chunk.as_mut_ptr();
+        let excepts: *mut u8 = except_chunk.as_mut_ptr();
+        
+        interface::fd_set_insert(inputs, serversockfd);
+        interface::fd_set_insert(inputs, filefd);
+        interface::fd_set_insert(outputs, filefd);
+
+        assert_eq!(interface::fd_set_check_fd(inputs, serversockfd), true);
+        assert_eq!(interface::fd_set_check_fd(inputs, filefd), true);
+        assert_eq!(interface::fd_set_check_fd(outputs, filefd), true);
 
         assert_eq!(cage.fork_syscall(2), 0);
         assert_eq!(cage.fork_syscall(3), 0);
 
         assert_eq!(cage.close_syscall(clientsockfd1), 0);
         assert_eq!(cage.close_syscall(clientsockfd2), 0);
 
+        let barrier = Arc::new(Barrier::new(3));
```

Comment:
a brief comment explaining the barrier would be helpful for people looking at this in the future